src/app/src/main/kotlin/org/imperial/mrc/hint/APIClient.kt

```diff
@@ -0,0 +1,33 @@
+package org.imperial.mrc.hint
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.github.kittinunf.fuel.core.Response
+import com.github.kittinunf.fuel.httpPost
+import org.springframework.context.annotation.Configuration
+
+interface APIClient {
+    fun validate(path: String, type: FileType): Response
+}
+
+@Configuration
+open class HintAPIClient(
```

Comment:
I wonder whether this class should call the .asResponseEntity() method on its responses itself, and return spring ResponseEntities rather than Fuel responses - Fuel is really an internal implementation detail of the HintAPIClient so it might be nicer to keep everything outside it in Spring terms?

Reply:
Seems reasonable